### PR TITLE
chore(api): add proper RBAC for api to reduce access

### PR DESCRIPTION
**Pull request by MatoPlus**
### Notion ticket link

[Patch All API Endpoints for minimal access - RBAC](https://www.notion.so/uwblueprintexecs/Patch-All-API-Endpoints-for-minimal-access-RBAC-5082191ef43149baa19d92f2f82f9b8f)

### Implementation description

- For all API points, we are ensuring that RBAC checks are performed when appropriate

### Steps to test

1. All frontend pages using affected API does not break if used as intended

### What should reviewers focus on?

- Any missing RBACs
- added RBAC logic implementation

### Checklist

-   [X] My PR name is descriptive and in imperative tense
-   [X] I have run the linter
-   [X] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/api/class/index.ts [hunk 1]
```diff
@@ -5,21 +5,29 @@ import { ClassInput, ClassTranslationInput } from "@models/Class";
...
  * @param req API request object
  * @param res API response object
  */
 export default async function handle(req: NextApiRequest, res: NextApiResponse): Promise<void> {
+    const session = await getSession({ req });
     switch (req.method) {
         case "GET": {
             const { id: programId, archived } = req.query;
 
+            const archivedParsed = Boolean(JSON.parse((archived as string) || "false"));
```
Comment (feng-eric) [inline pages/api/class/index.ts:22]: @MatoPlus Can you explain this part to me? How come we're doing this before we check `!isAdmin(session)`?

File: pages/api/user/index.ts [hunk 2]
```diff
@@ -73,13 +74,8 @@ export default async function handle(req: NextApiRequest, res: NextApiResponse):
             ResponseUtil.returnOK(res, updatedUser);
             break;
         }
-        case "DELETE": {
-            // TODO:
-            break;
-        }
         default: {
             const allowedHeaders: string[] = ["GET", "POST", "PUT", "DELETE"];
```
Comment (feng-eric) [inline pages/api/user/index.ts]:
```suggestion
            const allowedHeaders: string[] = ["GET", "POST", "PUT"];
```

File: pages/api/class/index.ts (hunk 1 above)
Comment (MatoPlus) [reply pages/api/class/index.ts:22]: My understanding is that the route is open to the public, but archived items requires admin access. 

quoting the comment: "get archived should be admin only"

File: pages/api/user/index.ts (hunk 2 above)
Comment (MatoPlus) [reply pages/api/user/index.ts]: good catch : )

File: pages/api/class/index.ts (hunk 1 above)
Comment (feng-eric) [reply pages/api/class/index.ts:22]: Might be good to leave a comment here or perhaps make sure this is documented somewhere along with which other routes should be private

**Review by feng-eric — APPROVED** [review]

File: pages/api/class/index.ts (hunk 1 above)
Comment (MatoPlus) [reply pages/api/class/index.ts:22]: i'll make the comment a bit more intuitive, we're just following the least privilege access principle for our api routes.

**Pull request merged**